analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -0,0 +1,290 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer.managers
+
+import ch.frankel.slf4k.*
+
+import com.fasterxml.jackson.databind.JsonNode
+
+import com.here.ort.analyzer.AbstractPackageManagerFactory
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.Identifier
+import com.here.ort.model.Package
+import com.here.ort.model.PackageLinkage
+import com.here.ort.model.PackageReference
+import com.here.ort.model.Project
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.Hash
+import com.here.ort.model.Scope
+import com.here.ort.model.VcsInfo
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.jsonMapper
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.log
+import com.here.ort.utils.textValueOrEmpty
+
+import com.vdurmont.semver4j.Requirement
+import com.moandjiezana.toml.Toml
+
+import java.io.File
+import java.util.SortedSet
+
+class Cargo(
+    name: String,
+    analyzerRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analyzerRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Cargo>("Cargo") {
+        override val globsForDefinitionFiles = listOf("Cargo.toml")
+
+        override fun create(
+            analyzerRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) =
+            Cargo(managerName, analyzerRoot, analyzerConfig, repoConfig)
+    }
+
+    companion object {
+        private const val REQUIRED_CARGO_VERSION = "1.0.0"
+        private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
+        private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
+    }
+
+    override fun command(workingDir: File?) = "cargo"
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildStrict(REQUIRED_CARGO_VERSION)
+
+    private fun runMetadata(workingDir: File): String {
+        return run(workingDir, "metadata", "--format-version=1").stdout
+    }
+
+    private fun extractCargoId(node: JsonNode): String {
+        return node["id"].textValue()!!
+    }
+
+    private fun extractPackageId(node: JsonNode) = Identifier(
```

Comment:
Nit: Formatting should be like

    private fun extractPackageId(node: JsonNode) =
        Identifier(
            // ...
        )